gammapy/maps/wcsnd.py

```diff
@@ -688,3 +692,68 @@ def cutout(self, position, width, mode="trim"):
         data = self.data[cutout_slices]
 
         return self._init_copy(geom=geom, data=data)
+
+    def sample_position_energy(self, npred_map=None, n_events=None, random_state=0):
```

Comment:
Please rename the method to `.sample_coord()` and remove the `npred_map` argument.

As we attach the `.sample()` method to the `Map` class, we can just access `self.data` and assume it contains an `npred` map already.